### PR TITLE
[PHP 8.0] Make use of property promotion - step 1

### PR DESCRIPTION
**Pull request by TomasVotruba**

### CODE REVIEW

**Comment by TomasVotruba** [comment]
Ready to go here :+1:  the failing is coverage lag

File: app/bundles/CoreBundle/Event/CustomButtonEvent.php [hunk 1]
```diff
@@ -7,35 +7,23 @@
...
     /**
      * CustomButtonEvent constructor.
      *
      * @param null $item
      */
-    public function __construct($location, Request $request, array $buttons = [], $item = null)
+    public function __construct(/**
+     * Button location requested.
+     */
+    protected $location, Request $request, array $buttons = [], protected $item = null)
```
Comment (escopecz) [inline app/bundles/CoreBundle/Event/CustomButtonEvent.php]:
```suggestion
    public function __construct(protected $location, Request $request, array $buttons = [], protected $item = null)
```

File: app/bundles/CoreBundle/Event/StatsEvent.php [hunk 2]
```diff
@@ -79,22 +74,21 @@ class StatsEvent extends Event
...
      * StatsEvent constructor.
      *
      * @param int $start
      * @param int $limit
      */
-    public function __construct($table, $start, $limit, array $order, array $where, User $user)
+    public function __construct($table, $start, $limit, /**
+     * Array of order by statements.
+     */
+    protected array $order, array $where, protected User $user)
```
Comment (escopecz) [inline app/bundles/CoreBundle/Event/StatsEvent.php]:
```suggestion
    public function __construct($table, $start, $limit, protected array $order, array $where, protected User $user)
```

File: app/bundles/CoreBundle/Helper/Chart/ChartQuery.php [hunk 3]
```diff
@@ -68,12 +63,14 @@ class ChartQuery extends AbstractChart
      *
      * @param string|null $unit
      */
-    public function __construct(Connection $connection, \DateTime $dateFrom, \DateTime $dateTo, $unit = null)
+    public function __construct(/**
+     * Doctrine's Connetion object.
+     */
+    protected Connection $connection, \DateTime $dateFrom, \DateTime $dateTo, $unit = null)
```
Comment (escopecz) [inline app/bundles/CoreBundle/Helper/Chart/ChartQuery.php]:
```suggestion
    public function __construct(protected Connection $connection, \DateTime $dateFrom, \DateTime $dateTo, $unit = null)
```

File: app/bundles/CoreBundle/IpLookup/AbstractLookup.php [hunk 4]
```diff
@@ -53,16 +38,19 @@ abstract protected function lookup();
...
-    {
-        $this->cacheDir  = $cacheDir;
-        $this->logger    = $logger;
-        $this->auth      = $auth;
-        $this->config    = $ipLookupConfig;
-        $this->client    = $client;
+    public function __construct(
+        /**
+         * Authorization for lookup service.
+         */
```
Comment (escopecz) [inline app/bundles/CoreBundle/IpLookup/AbstractLookup.php]:
```suggestion
```

File: app/bundles/FormBundle/Collector/AlreadyMappedFieldCollector.php [hunk 5]
```diff
@@ -11,13 +11,12 @@
...
 
-    private CacheProviderInterface $cacheProvider;
-
-    public function __construct(CacheProviderInterface $cacheProvider)
-    {
-        $this->cacheProvider = $cacheProvider;
+    public function __construct(
+        // 5 hours
+        private CacheProviderInterface $cacheProvider
+    ) {
```
Comment (escopecz) [inline app/bundles/FormBundle/Collector/AlreadyMappedFieldCollector.php]: In this case the comment even doesn't belong to the property but to the const above it.
```suggestion
    public function __construct(private CacheProviderInterface $cacheProvider) {
```

File: app/bundles/CoreBundle/Validator/FileUploadValidator.php [hunk 6]
```diff
@@ -8,14 +8,12 @@
...
-
-    public function __construct(TranslatorInterface $translator)
-    {
-        $this->translator = $translator;
+    public function __construct(
+        /**
+         * @param TranslatorInterface $translator
+         */
+        protected TranslatorInterface $translator
+    ) {
```
Comment (escopecz) [inline app/bundles/CoreBundle/Validator/FileUploadValidator.php]:
```suggestion
    public function __construct(protected TranslatorInterface $translator) {
```

File: app/bundles/CoreBundle/Model/IteratorExportDataModel.php [hunk 7]
```diff
@@ -7,40 +7,27 @@
...
      * @param AbstractCommonModel<T> $model
      * @param array<mixed>           $args
      *
      * @template T of object
      */
-    public function __construct(AbstractCommonModel $model, array $args, callable $callback, bool $skipOrdering = false)
+    public function __construct(/**
+     * @template T of object
+     */
+    private AbstractCommonModel $model, private array $args, callable $callback, private bool $skipOrdering = false)
```
Comment (escopecz) [inline app/bundles/CoreBundle/Model/IteratorExportDataModel.php]:
```suggestion
    public function __construct(private AbstractCommonModel $model, private array $args, callable $callback, private bool $skipOrdering = false)
```

**Review by escopecz — CHANGES_REQUESTED** [review]
There is quite a lot of changes. I noticed only some weird comment placement. I suggest to remove them.

**Comment by TomasVotruba** [comment]
I've included all the suggested changed :+1:

**Review by escopecz — APPROVED** [review]

**Comment by TomasVotruba** [comment]
Awesome 😎😎😎

**Pull request merged**